hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HiveHoodieReaderContext.java

```diff
@@ -89,41 +84,17 @@ public class HiveHoodieReaderContext extends HoodieReaderContext<ArrayWritable>
   private final String recordKeyField;
 
   protected HiveHoodieReaderContext(HoodieFileGroupReaderBasedRecordReader.HiveReaderCreator readerCreator,
```

Comment:
reader context shouldn't be initialized with things specific to a particular fg. Got rid of some unused inputs, and moved the calculations for some things like tablename, partition cols, and recordkey field into the reader so that this class doesn't have big dependencies on things like metaclient etc.